apis/v1beta1/validation/httproute_test.go

```diff
@@ -1049,7 +1075,7 @@ func TestValidateHTTPRouteTypeMatchesField(t *testing.T) {
 				StatusCode: new(int),
 			},
 		},
-		errCount: 0,
+		errCount: 1,
```

Comment:
why has this changed ?

Reply:
So, as in this issue we need to validate that BackendRefs is not used when Redirect Filters  are used in httpRoute. 
In this [example](https://github.com/kubernetes-sigs/gateway-api/blob/main/apis/v1beta1/validation/httproute_test.go#L1041) as we can see that we are using Redirect Filters and then in the for loops we have [BackendRefs](https://github.com/kubernetes-sigs/gateway-api/blob/main/apis/v1beta1/validation/httproute_test.go#L1125) as well. 

After I added a check of it inside `validation/httproute.go`. The test started failing so I made the `errCount` as 1 .

Reply:
ah thanks for the clarification, didnt go through at the entire file earlier 🙈 
would be good to add a positive test case, maybe as a new standalone function that ensures thats when the Redirect filter exists and no backendRefs are set, validation is successful

Reply:
was referring to a standalone alone test `TestValidateRequestRedirectFiltersWithNoBackendRef`
anyways this is a good to have and is non blocking